packages/babel-generator/src/node/parentheses.js

```diff
@@ -80,55 +75,35 @@ export function Binary(node: Object, parent: Object): boolean {
 }
 
 export function BinaryExpression(node: Object, parent: Object): boolean {
-  if (node.operator === "in") {
+  if (
+    node.operator === "in" &&
     // let i = (1 in []);
-    if (t.isVariableDeclarator(parent)) {
-      return true;
-    }
-
+    (t.isVariableDeclarator(parent) ||
     // for ((1 in []);;);
-    if (t.isFor(parent)) {
-      return true;
-    }
+    t.isFor(parent))
+  ) {
+    return true;
   }
 
   return false;
 }
 
 export function SequenceExpression(node: Object, parent: Object): boolean {
-  if (t.isForStatement(parent)) {
+
+  if (
     // Although parentheses wouldn"t hurt around sequence
     // expressions in the head of for loops, traditional style
     // dictates that e.g. i++, j++ should not be wrapped with
     // parentheses.
-    return false;
-  }
-
-  if (t.isExpressionStatement(parent) && parent.expression === node) {
-    return false;
-  }
-
-  if (t.isReturnStatement(parent)) {
-    return false;
-  }
-
-  if (t.isThrowStatement(parent)) {
-    return false;
-  }
-
-  if (t.isSwitchStatement(parent) && parent.discriminant === node) {
-    return false;
-  }
-
-  if (t.isWhileStatement(parent) && parent.test === node) {
-    return false;
-  }
-
-  if (t.isIfStatement(parent) && parent.test === node) {
-    return false;
-  }
-
-  if (t.isForInStatement(parent) && parent.right === node) {
+    t.isForStatement(parent) ||
+    t.isExpressionStatement(parent) && parent.expression === node ||
+    t.isReturnStatement(parent) ||
+    t.isThrowStatement(parent) ||
+    t.isSwitchStatement(parent) && parent.discriminant === node ||
+    t.isWhileStatement(parent) && parent.test === node ||
+    t.isIfStatement(parent) && parent.test === node ||
+    t.isForInStatement(parent) && parent.right === node
```

Comment:
I personally don't think this improves readability. I want to hear other thought about that.